### PR TITLE
Easier Settings for Tray publisher addon

### LINKED ISSUE #43: Easier Settings for Tray publisher addon
### Is there an existing issue for this?

- [x] I have searched the existing issues and added correct labels.

### Please describe the feature you have in mind and explain what the current shortcomings are?

Settings are hard to read and modify. 
- `simple create` contains many items and it's hard to go through them.

### Suggested implementation?

.

### Describe alternatives you've considered:

.

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
resolve #43 

This PR only changes the settings UI. 

1. Make items of `Simple creators` easier to read. 
by un expanding the item groups. 
image
2. Add various enhancements to `IngestCSV` settings
    - Columns settings are expanded, making it easier to have a quick look through the whole section
    image
    - Doing the same with representations
    image
    - Rearrange the `Folder creation config` setting and change the layout of folder types to compact (so it looks like a table of regexes)
     image

## Testing notes:
1. Everything should work as before.

### CODE REVIEW

**Comment by iLLiCiTiT** [comment]
This PR changig a lot of things at the same time.

I don't think we should put simple and editorial creators under `create` TBH. That's my point of view, so we'd need some other person to decide that...

File: server/settings/conversion.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+def _convert_simple_creators_0_2_9(overrides):
+    if "simple_creators" not in overrides:
+        return
+    
+    if overrides.get("create") is None:
+        overrides.update({
+            "create": {}
+        })
+    
+    overrides["create"]["simple_creators"] = overrides.pop("simple_creators")
```
Comment (iLLiCiTiT) [inline server/settings/conversion.py:13]:
```suggestion
    create_overrides = overrides.setdefault("create", {})
    create_overrides["simple_creators"] = overrides.pop("simple_creators")
```

File: server/settings/conversion.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+def _convert_editorial_creators_0_2_9(overrides):
+    if "editorial_creators" not in overrides:
+        return
+    
+    if overrides.get("create") is None:
+        overrides.update({
+            "create": {}
+        })
+    
+    overrides["create"]["editorial_creators"] = overrides.pop("editorial_creators")
```
Comment (iLLiCiTiT) [inline server/settings/conversion.py:25]:
```suggestion
    create_overrides = overrides.setdefault("create", {})
    create_overrides["editorial_creators"] = overrides.pop("editorial_creators")
```

File: server/settings/creator_plugins.py [hunk 3]
```diff
@@ -164,7 +175,20 @@ class FolderCreationConfigModel(BaseSettingsModel):
         title="Enabled folder creation",
         default=False,
     )
-
+    folder_create_type: str = SettingsField(
```
Comment (iLLiCiTiT) [inline server/settings/creator_plugins.py:178]: @jakubjezek001 @robin-ynput is changing of these attributes order ok for you?

File: server/settings/creator_plugins.py [hunk 4]
```diff
@@ -231,13 +238,23 @@ class TrayPublisherCreatePluginsModel(BaseSettingsModel):
...
     )
+    editorial_creators: TraypublisherEditorialCreatorPlugins = SettingsField(
+        title="Editorial Creators",
+        default_factory=TraypublisherEditorialCreatorPlugins,
+    )
     IngestCSV: IngestCSVPluginModel = SettingsField(
         title="Ingest CSV",
         default_factory=IngestCSVPluginModel
     )
+    simple_creators: list[SimpleCreatorPlugin] = SettingsField(
```
Comment (iLLiCiTiT) [inline server/settings/creator_plugins.py:249]: If we'll move it here, it should be at first place.

File: server/settings/simple_creators.py [hunk 5]
```diff
@@ -1,13 +1,16 @@
+from pydantic import root_validator
+
 from ayon_server.settings import BaseSettingsModel, SettingsField
 
 
 class SimpleCreatorPlugin(BaseSettingsModel):
-    _layout = "expanded"
-    product_type: str = SettingsField("", title="Product type")
+
+    name: str = SettingsField(title="Name", disabled=True)
```
Comment (iLLiCiTiT) [inline server/settings/simple_creators.py:8]: What is the name for?

File: server/settings/simple_creators.py [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
+    label: str = SettingsField("", title="Label")
     # TODO add placeholder
     identifier: str = SettingsField("", title="Identifier")
-    label: str = SettingsField("", title="Label")
     icon: str = SettingsField("", title="Icon")
+    product_type: str = SettingsField("", title="Product type")
```
Comment (iLLiCiTiT) [inline server/settings/simple_creators.py:13]: Why did you move the most important thing at the end of the item?

File: server/settings/simple_creators.py (hunk 5 above)
Comment (MustafaJafar) [reply server/settings/simple_creators.py:8]: yes, it's to show the name at the top

File: server/settings/simple_creators.py (hunk 6 above)
Comment (MustafaJafar) [reply server/settings/simple_creators.py:13]: Because this is more familiar to me. 
when implementing a creator, we first set the label, identifier icon and then add the product type.

Comment (MustafaJafar) [reply server/settings/simple_creators.py:13]: although the product type is more important to AYON. label can be more important for users. because they will see it in the publisher UI.
image

Comment (iLLiCiTiT) [reply server/settings/simple_creators.py:13]: The only required field is product type and extensions, the others are optional...

Comment (MustafaJafar) [reply server/settings/simple_creators.py:13]: let me try one more time.

File: server/settings/creator_plugins.py (hunk 4 above)
Comment (MustafaJafar) [reply server/settings/creator_plugins.py]: I'll make it first  https://github.com/ynput/ayon-traypublisher/pull/44/commits/b7dbe8b6f9534cda27e00d39d98bab228e3c3b4b

and we can decide if we want to keep them or revert the changes.

File: server/settings/creator_plugins.py (hunk 3 above)
Comment (robin-ynput) [reply server/settings/creator_plugins.py:178]: I'm not sure tbh.. if I understand correctly this change, we go from:
```
folder regexes
default folder type
--
task regexes
default task type
```
to 
```
default folder type
default task type
folder regexes
task regexes
```
Purely personal preferences here, but I think I'd prefer keeping them grouped per folder/task instead. 
Not a biggie if other people disagree.

**Review by robin-ynput — APPROVED** [review]
I appreciate most of this can be seen as personal preferences.

But I'd say I'm happy with those changes:
🟢 Ingest CSV column expanded
🟢 Ingest CSV representation expanded

Slightly less sure about those ones:
🟠 Folder/Task Regexes order switch - _current order makes more sense to me_
🟠 Grouping settings under `Create` - _looks fine on paper but I haven't played much with those option so it's hard to give a great feedback here_

I'm giving an approval here to not slow things down, I feel once discussions will happen I this will just be matter of finding a proper compromise.

**Comment by MustafaJafar** [comment]
> 🟠 Folder/Task Regexes order switch - _current order makes more sense to me_

@robin-ynput , I admit the changes _mostly_ came from my personal preference. 
But, honestly, the original settings are uncomfortable to my eyes 😅, they go like: 
```
item
expandable folder
item
expandable folder
```
image

I've here few suggestions, please let me know which one do you prefer: 
| Current status of this PR | Alternative suggestion |
|--|--|
| image | image |

**Comment by robin-ynput** [comment]
> > 🟠 Folder/Task Regexes order switch - _current order makes more sense to me_
> 
> @robin-ynput , I admit the changes _mostly_ came from my personal preference. But, honestly, the original settings are uncomfortable to my eyes 😅, they go like:
> 
> ```
> item
> expandable folder
> item
> expandable folder
> ```
> 
> image
> 
> I've here few suggestions, please let me know which one do you prefer:
> 
> Current status of this PR	Alternative suggestion
> image	image

Important thing imo is really to keep Folder/Task grouping. I think the alternative suggestion is great !!

File: server/settings/creator_plugins.py (hunk 3 above)
Comment (MustafaJafar) [reply server/settings/creator_plugins.py:178]: Thank you for pointing that out. 
The original order is used while enhancing it with a section argument. 
https://github.com/ynput/ayon-traypublisher/pull/44/commits/2b496aa910cd3d1fb4a9543bdfbb86bb2df25069

**Comment by MustafaJafar** [comment]
should this PR get a `bump minor` label?

File: server/settings/simple_creators.py (hunk 5 above)
Comment (iLLiCiTiT) [reply server/settings/simple_creators.py:8]: BTW `name` is also auto-validated on frontend to be unique for all items in the model. So until frontend does support all features we need, then this won't happen.

**Comment by iLLiCiTiT** [comment]
> should this PR get a bump minor label?

You should split the PR to multiple PRs. There are different major changes, some of them cannot be merged and some of them should be approved by different people.

File: server/settings/simple_creators.py (hunk 6 above)
Comment (MustafaJafar) [reply server/settings/simple_creators.py]: Reverted the change in order.
https://github.com/ynput/ayon-traypublisher/pull/44/commits/03df7b381548c5ca73b89700da93f3e833b6e9a1

**Review by robin-ynput — APPROVED** [review]

File: server/settings/simple_creators.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 
 class SimpleCreatorPlugin(BaseSettingsModel):
-    _layout = "expanded"
```
Comment (iLLiCiTiT) [inline server/settings/simple_creators.py:5]: This really should stayed there.

Comment (MustafaJafar) [reply server/settings/simple_creators.py:5]: IMO, Expanded layout in this setting  is very confusing. it's very hard for me to go through it.
image

Comment (MustafaJafar) [reply server/settings/simple_creators.py:5]: I tried updating the setting group name with `label` but that was not recommended because admins are allowed to leave the label blank. 

Also, I tried updating the setting group name with `product_type` but AYON server ignored the name in favor of the label. 
https://discord.com/channels/517362899170230292/773127690453516298/1313448327903445012

Comment (iLLiCiTiT) [reply server/settings/simple_creators.py:5]: I know you've tried to change, but then reverted, so this should be reverted too. It was bad before, now it is worse...

Comment (MustafaJafar) [reply server/settings/simple_creators.py:5]: I'll create another PR this afternoon.
